src/yunohost/certificate.py

```diff
@@ -837,7 +843,10 @@ def _dns_ip_match_public_ip(public_ip, domain):
 
 def _domain_is_accessible_through_HTTP(ip, domain):
     try:
-        requests.head("http://" + ip, headers={"Host": domain})
+        requests.head("http://" + ip, headers={"Host": domain}, timeout=30)
+    except Timeout as e:
+        # XXX what should we do here? retry?
```

Comment:
30 sec also seems a lot to me. 5 to 10 should be more than enough. Here we are trying to contact ourself, and we just want to detect the case where it does not work at all (due to hairpinning or other weird firewall setups).